### ISSUE TITLE
typeahead not displaying sometimes 2.2.1

### ISSUE THREAD

**Issue by bbloomf** [mgcrea/angular-strap#1588]
The typeahead directive isn't working properly when it gets the focus while the input box is still empty, at least when min-length is set to 1; I haven't tried it with min-length=0.  The issue is that when it is positioning the tooltip, the tooltip is not visible and so its offsets are all set to 0.

**Comment by metronomx** [comment]
A quick solution:

Add following some rule:
`input + .typeahead.ng-hide { display: block !important; visibility: hidden !important; overflow: hidden; }`

Elements with `visibility: hidden` still have size, and this will allow typeahead popup to calculate correct values for postioning and size.
Optionally you can add `opacity: 0;` if needed.

**Comment by ansonkao** [comment]
Confirmed that @metronomx 's fix works! Thanks!
Though not a long term solution

**Comment by ansonkao** [comment]
Note that @metronomx 's fix does not work when data-container="body"

**Comment by metronomx** [comment]
Correct. If data-container is "body", then fix would be

`body > .typeahead.ng-hide { display: block !important; visibility: hidden !important; overflow: hidden; }`

or, in general case

`[whatever is in data-container] > .typeahead.ng-hide { display: block !important; visibility: hidden !important; overflow: hidden; }`

**Comment by gligoran** [comment]
I've hit the same problem. As @bbloomf said, it seems to be tied to the ```min-length``` property. If it is set to 0 it works, as it should (it displays the list immediately). If set to anything else or leaving the default value of 1, by not supplying this attribute, it works strangely. It will only display the list, if you focus the input field, once its contents already meets the minimum length requirement. But it won't display it if at the time of focusing, the contents didn't meet the requirement. I believe 0 isn't an exception, but rather an always-true type situation.

I've made this plunk, which better illustrates this (especially playing with the last two in the way I described above): http://plnkr.co/voGLhx4wgw3mWNuYx6nB

**Comment by subu28** [comment]
It is not linked to 'min-length' as @gligoran's suggests. This issue occurs anytime a situation arrises where the popup is not displayed immediately when triggered.

In @gligoran's plunk select a object in any of the typeahead, click elsewhere and then come back to the typeahead and edit its content. The popup does not appear forcing user to click elsewhere and come back to the typeahead to trigger the popup.

**Comment by dlsso** [comment]
I am having this problem as well. Seems to have been introduced in 2.2.0, #1474 may be the same issue. Will try css workaround for now.

**Comment by daniel-nalbach** [comment]
+1 for CSS workaround

Problem occurs for me on Angular 1.2.15. Is this problem limited to 1.2.x?

**Comment by gligoran** [comment]
@daniel-nalbach, no it's not. I'm using Angular 1.3.15 and it's still there.

**Comment by gligoran** [comment]
I think I found another example of this: http://plnkr.co/edit/khA9V1wx45hITdf4Q0IO?p=preview

In this demo I'm moving states (US, not some kind of control states) between two lists. When I click the first type-ahead the options display nicely. I select one and click add, which removes it from the first list and adds it to another. Now if I click in the second typeahead input box no list is displayed even though, minLength is 0. As soon as I type a single letter that was in the selected state's name, I get the typeahead list. The problem, however seems to be with updating the available options. If now, after managing to drag out the options in second list, I go back to the first one, search, select and add another state, then go back to the second typeahead, only the firstly added item is displayed. Again, if I type anything, the options are updated.

**Comment by andras-tim** [comment]
Hi all!

I'm spent lot of time on [the official Plunker site of typeahead](http://mgcrea.github.io/angular-strap/#/typeaheads), and tried to made [my POC code](https://github.com/andras-tim/poc-for-everything/commits/master/web/angular-strap2.2) to works like the official. I did not know why works the official, and why not my same POC site.

**I have found the cause of this tricky problem (and workaround). The ``angular-animate`` was missed from my POC for auto-complete woks.** Why? I don't know, but after I added and enabled this module to my POC, then it was started to work as I want.

Here is a bit modified Plunker site of the official typeahead example for check my thesis:
http://plnkr.co/edit/v2BdeoqNFLJV4QRkXYxC?p=preview
